### PR TITLE
Move TechEmpowerBenchmark Server to this repo

### LINKED ISSUE #615: Move TechEmpowerBenchmark Server to this repo
Currently, the ZIO Http Server code for the benchmarks is written in — https://github.com/dream11/FrameworkBenchmarks

So any kind of breaking changes in ZIO Http, requires us to make a code commit in the above repository, OR wait until the CI fails with a compile error. Moreover, any kind of server optimization that requires user-land code changes, also resides in the `dream11/FrameworkBenchmarks` repository, making it harder to test optimizations.

**Proposal**
Keep the techempower benchmarking code in this (`dream11/zio-http` repository (preferably under the `/benchmark` project). 

**Advantages**
Easier to make changes to local code and test improvements in performance etc.

**Disadvantage**
We could directly send a PR to the `TechEmpower/FrameworkBenchmarks` before because the code resided inside the forked repo `Dream11/FrameworkBenchmarks`. This way, however, we will have to copy the server code and then send a PR. I don't think this is a big issue since the updates on `TechEmpower/FrameworkBenchmarks` are going to be very few (1 in a few months).

### PR DESCRIPTION
**Pull request by amitksingh1490**
To resolve #615

### CODE REVIEW

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 440239.23**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 430751.65**

File: zio-http-benchmarks/src/main/scala/Main.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import io.netty.handler.codec.http.HttpHeaderNames
+import io.netty.util.AsciiString
+import zhttp.http.{Response, _}
+import zhttp.service.server.ServerChannelFactory
+import zhttp.service.{EventLoopGroup, Server}
+import zio.{App, ExitCode, URIO}
+
+case class Message(message: String)
+
+object Main extends App {
```
Comment (tusharmath) [inline zio-http-benchmarks/src/main/scala/Main.scala]: - Move this to `Example` project
- Rename to `PlainTextBenchmarkServer`
- Add a scala-doc to explain what this server does.

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 437573.74**

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -251,6 +251,8 @@ jobs:
 
       - id: result
         run: |
+          rm ./FrameworkBenchMarks/frameworks/Scala/zio-http/src/main/scala/Main.scala
+          mv ./zio-http/example/src/main/scala/example/PlainTextBenchmarkServer.scala ./FrameworkBenchMarks/frameworks/Scala/zio-http/src/main/scala/Main.scala
```
Comment (tusharmath) [inline .github/workflows/ci.yml]: should this be a copy command?

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 439073.22**

**Review by tusharmath — APPROVED** [review]

**Pull request merged**